platform/o.n.bootstrap/src/org/netbeans/ModuleManager.java

```diff
@@ -1793,6 +1838,13 @@ private boolean searchForPossibleEager(Set<Module> willEnable, Set<Module> still
                 continue;
             }
             if (m.isEager()) {
+                if (m.getFragmentHostCodeName() != null) {
```

Comment:
Prevent eagers to 'just enable' if its host is not ... Maybe redundant, since if the `eager` uses Module-Dependency on its host, it would work as well.